surpress webkit.RefCntblBaseVirtualDtor warnings for ArrayBufferView

colelao opened the pull request:
#### 7025aef8e42637810c4eaf7ebc7b18386c658235
<pre>
surpress webkit.RefCntblBaseVirtualDtor warnings for ArrayBufferView
<a href="https://bugs.webkit.org/show_bug.cgi?id=296889">https://bugs.webkit.org/show_bug.cgi?id=296889</a>
<a href="https://rdar.apple.com/157492341">rdar://157492341</a>

Reviewed by Yusuke Suzuki.

This patch makes ArrayBufferView inherit from the NoVirtualDestructorBase class to surpress the Safer Cpp warnings.

* Source/JavaScriptCore/runtime/ArrayBufferView.h:

Canonical link: <a href="https://commits.webkit.org/298577@main">https://commits.webkit.org/298577@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b591c909968a800c3b584d3dc8a0ef055efaadee

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115291 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34987 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25487 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121389 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66064 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35644 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43557 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87589 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/42418 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118239 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28417 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103492 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67985 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27580 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21615 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65044 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/107535 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97802 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21729 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124560 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/113833 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42237 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31615 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96382 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42606 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99682 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96167 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24579 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41391 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19241 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/38179 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42116 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47669 "Built successfully") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/138156 "Build was cancelled. Recent messages:Printed configuration; Running apply-patch; Failed to checkout and rebase branch from PR 48912") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41633 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/138156 "Build was cancelled. Recent messages:Printed configuration; Running apply-patch; Failed to checkout and rebase branch from PR 48912") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44957 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43359 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (commented):
Is it guaranteed that Empty Base Optimization kicks in with this class?

Constellation reviewed (approved):
r=me if EBO is working for this class in the current form.

colelao committed:
surpress webkit.RefCntblBaseVirtualDtor warnings for ArrayBufferView
https://bugs.webkit.org/show_bug.cgi?id=296889
rdar://157492341

Reviewed by Yusuke Suzuki.

This patch makes ArrayBufferView inherit from the NoVirtualDestructorBase class to surpress the Safer Cpp warnings.

* Source/JavaScriptCore/runtime/ArrayBufferView.h:

Canonical link: https://commits.webkit.org/298577@main
